src/core/sig_state.ml

```diff
@@ -196,6 +194,19 @@ let open_sign : sig_state -> Sign.t -> sig_state = fun ss sign ->
   in
   {ss with in_scope; builtins; unops; binops; pp_hints}
 
+(** [of_sign sign] creates a state from the signature [sign] with ghost
+    signatures opened. *)
+let of_sign : Sign.t -> sig_state = fun sign ->
+  let empty =
+    { signature = sign; in_scope = StrMap.empty; aliases = StrMap.empty
+    ; path_map = PathMap.empty; builtins = StrMap.empty; unops = StrMap.empty
+    ; binops = StrMap.empty; pp_hints = SymMap.empty }
+  in
+  open_sign empty Unif_rule.sign
+
+(** [empty] state. *)
+let empty : sig_state = of_sign (create_sign [])
```

Comment:
You can now replace create_sign [] by dummy. Perhaps the empty state should be rename into dummy too.

Reply:
Ah no because `create_sign` sets the dependencies on `Unif_rule`

Reply:
And I prefer leaving `empty` as a name for "ready-to-use" elements and `dummy` for not usable elements: the dummy signature needs to be initialised with ghost signatures and path before being used but the empty `sig_state` can be nominally used.

Reply:
But this empty/dummy sig_state is not usable ([] is not a valid path). It is used in print only. That's why you can rename it into dummy and use it in of_sign.

Reply:
If it's used in `Print` only, don't we want to define it there?